src/sortingtraining/sortalgorythms/BucketSort.java

```diff
@@ -9,42 +9,57 @@
 
         int[] numbsToSort = Arrays.copyOf(numbers, numbers.length);
 
+        if (numbsToSort.length <= 1) {
+            System.out.println("WARN: There is nothing to sort!!! \n");
+            return numbsToSort;
+        }
+
         int largest = Integer.MIN_VALUE;
         int lowest = Integer.MAX_VALUE;
 
         //looking for min and max value
-        for (int n : numbsToSort) {
-            if (n >largest) {
-                largest = n;
+        for (int i : numbsToSort) {
+            if (i >largest) {
+                largest = i;
             }
-            if (n < lowest) {
-                lowest = n;
+            if (i < lowest) {
+                lowest = i;
             }
         }
 
+        boolean negatives = false;
+
+        if (lowest < 0) {
+            negatives = true;
+        }
+
         // if we have negative integers, adding the lowest (abs) one to every record in array, to eliminate negatives
-        if  (lowest < 0) {
+        if  (negatives) {
             lowest = Math.abs(lowest);
-            for (int n = 0; n < numbsToSort.length; n++) {
-                numbsToSort[n] += lowest;
+            for (int j = 0; j < numbsToSort.length; j++) {
+                numbsToSort[j] += lowest;
             }
             largest += lowest;
-
         }
 
         //creating our helper array - bucket array
         int[] buckets = new int[largest + 1];
 
         //populating helper array
-        for (int n : numbsToSort) {
-            buckets[n] = buckets[n] + 1;
+        for (int k : numbsToSort) {
+            buckets[k] = buckets[k] + 1;
         }
 
         //finally printing sorted array, and decreasing by abs value of lowest, to retrieve original numbers
         int i = 0;
-        for (int n = 0; n < buckets.length; n++) {
-                for (int j = 0; j < buckets[n]; j++) {
-                    numbsToSort[i] = n - lowest;
+        for (int l = 0; l < buckets.length; l++) {
```

Comment:
You can assign multiply variables here, at the _initialization_ part of loop, try `for (int i = 0, j = 0; j < buckets.length; j++) {`

Reply:
Wow, it is working, i didnt knew about this 'trick' ! :)